### ISSUE TITLE
Debian/Ubuntu package versions aren't monotonic across OS releases

### ISSUE THREAD

**Issue by ilmari** [matrix-org/synapse#13553]
### Description

The versions of the Debian/Ubuntu packages use the codename in the build version, which aren't necessarily monotonic. This means that when upgrading from one OS version to the next, Synapse won't be upgraded to the matching version unless the codenames happen to be in alphabetical order (which they usually are on Ubuntu, but not on Debian).

### Steps to reproduce

- Install `matrix-synapse-py3` from packages.matrix.org on Debian Buster
- Upgrade to Bullseye, adjusting the codename in `/etc/apt/sources.list.d/matrix-org.list`
- Note that `matrix-synapse-py3` is still the Buster version

To fix this, use the OS version number in the build number, instead of the codename, e.g.
 `1.65.0-1.mxo11+1` for Debian 11 (Bullseye), and `1.65.0-1.mxo22.04+1` for Ubuntu 22.04 (Jammy Jellyfish).

### Homeserver

N/A

### Synapse Version

N/A

### Installation Method

Debian packages from packages.matrix.org

### Platform

Debian/Ubuntu

### Relevant log output

```shell
N/A
```

**Comment by richvdh** [comment]
Yup. I think fixing this just requires changing [this line](https://github.com/matrix-org/synapse/blob/v1.65.0/docker/build_debian.sh#L28) to use `$(lsb_release -rs)` instead of `$DIST`.

**Comment by richvdh** [comment]
oh, we'd want to include "debian" or "ubuntu" in there too.

**Comment by ilmari** [comment]
Richard van der Hoff ***@***.***> writes:

> oh, we'd want to include "debian" or "ubuntu" in there too.

I don't think that's necessary, since they have different version
formats, and the packages are in different archive suites.

I was basing the version format on the one used by apt.postgresql.org
(which is managed by the Debian PostgreSQL maintainer, who knows what
they're doing), and they support both Debian and Ubuntu without the
distro name in the version.

**Comment by richvdh** [comment]
not necessary, just clearer imho.

(why `mxo`?)

**Comment by ilmari** [comment]
Richard van der Hoff ***@***.***> writes:

> not necessary, just clearer imho.
>
> (why `mxo`?)

Short for matrix.org, to indicate the origin of the packge.  Similarly
to how apt.postgresql.org uses `pgdg` (PostgreSQL Global Development
Group) and backports.debian.org uses `bpo` (formerly backports.org).
